### ISSUE TITLE
Grademark pencil not clickable

### ISSUE THREAD

**Issue by ianmcn** [turnitin/moodle-mod_turnitintooltwo#11]
On version 2015040110, Moodle 2.8.
Pencil icon for uploaded assignments is not clickable, it is only clickable before an assignment is uploaded for creating grading template, after that it's no longer clickable, you can only access grademark through clicking the paper title or similarity percentage. This is the same on both Chrome and IE.

**Comment by amclarty** [comment]
Hi @ianmcn,

I've tested your issue on the latest versions of Safari, Chrome and Firefox and the Grademark icon can be used to create an empty submission and open the Document Viewer after submission.

Are you using any sort of ad blocker?

Can you let me know what operating system you're using and what versions of Chrome and IE? 

Thanks,

Alex.

**Comment by ianmcn** [comment]
﻿﻿Hi. I'm at home right now using Firefox 39 on OSX 10.11. At work (where I reported it) I am on Windows 7, latest version of Chrome and IE11 (I think). In every environment/browser I get this issue. The button is clickable when there isn't an assignment or grading template (and clicking generates the grading template successfully), but when there is already a submission or grading template, it's not clickable. I noticed today in the javascript console the following warning comes up when clicking one of these pencils:
Empty string passed to getElementById().

**Comment by amclarty** [comment]
Have you purged caches on your Moodle environment and your browser?

I've seen unusual behaviour with JavaScript if files are cached and a new version is installed.

**Comment by ianmcn** [comment]
Yes, we've purged caches. I've just delved into the javascript and markup more and found that the div is missing the required class "grademark_open". If I add this, it works. 

This is the markup in question:
Doesn't work:
```
<div title="[url removed]" id="grademark_[id removed]" class="box grademark_opengraded_warning">
```
Works:
```
<div title="[url removed]" id="grademark_[id removed]" class="box grademark_opengraded_warning grademark_open">
```

**Comment by ianmcn** [comment]
In fact looking more closely it appears that two class names have been concatenated together. Instead of "grademark_opengraded_warning" it should be "grademark_open graded_warning".

**Comment by ianmcn** [comment]
I think I can see why you don't experience the same behaviour. Looking at line 1225 of turnitintooltwo_view.class.php it seems the "graded_warning" class is only applied when resubmissions and late submissions are allowed. If that class isn't applied the issue won't occur. Can you test again with an assignment where resubmissions and late submissions are allowed?

**Comment by amclarty** [comment]
Sounds like you've identified the issue. I'll confirm this tomorrow and we'll get it fixed asap. Thanks for your feedback.

**Comment by amclarty** [comment]
@ianmcn, thanks for taking the time to find the root of the problem. I've put a fix in. We'll get this merged as soon as possible.

**Comment by ianmcn** [comment]
Hi @amclarty when will this be in a release?

**Comment by cwestgarth** [comment]
Hi @ianmcn, 

We're looking to do a small release at the end of the month. 

Thanks

**Comment by ianmcn** [comment]
OK, I guess I'll just manually apply the patch for now. It's quite a serious issue though, for institutions that allow resubmissions it completely breaks the direct method of opening grademark. I hoped a fix would be out sooner.

**Comment by jmcgettrick** [comment]
This has now been released in the master branch for version 2015040111.
